src/StructuredContentList.js

```diff
@@ -1,307 +1,336 @@
-        let queryParams = [];
+            return filter;
+          })
+        : []
+    };
 
-        if (activeFilters) {
-            queryParams.push(`filters=${activeFilters}`);
-        }
-        if (appliedFilterQuery) {
-            queryParams.push(`$filter=${appliedFilterQuery}`);
-        }
+    this.onChange = this.onChange.bind(this);
+    this.getFilterOptions = this.getFilterOptions.bind(this);
+    this.getActiveFilters = this.getActiveFilters.bind(this);
+    this.getAppliedFiltersQuery = this.getAppliedFiltersQuery.bind(this);
 
-        return queryParams.join('&');
-    }
+    this.onLoadMore = this.onLoadMore.bind(this);
+  }
 
-    buildQueryString() {
-        const queryString = this.buildFilterQueryString();
-        const activePage = this.state.activePage;
+  buildFilterQueryString() {
+    const activeFilters = this.getActiveFilters();
+    const appliedFilterQuery = this.getAppliedFiltersQuery();
+    let queryParams = [];
 
-        return queryString
-            ? `?page=${activePage}&${queryString}`
-            : `?page=${activePage}`;
+    if (activeFilters) {
+      queryParams.push(`filters=${activeFilters}`);
     }
-
-    componentDidMount() {
-        this.getBlogEntries(() => {
-            this.setState({
-                isInitialized: true
-            });
-        });
+    if (appliedFilterQuery) {
+      queryParams.push(`$filter=${appliedFilterQuery}`);
     }
 
-    getActiveFilters() {
-        // filters are the distinct fields being returned, if there are some in the app, these should always be added to the query string
-        return this.state.filters.map(filter => filter.field);
-    }
+    return queryParams.join("&");
+  }
+
+  buildQueryString() {
+    const queryString = this.buildFilterQueryString();
+    const activePage = this.state.activePage;
+
+    return queryString
+      ? `?page=${activePage}&${queryString}`
+      : `?page=${activePage}`;
+  }
+
+  componentDidMount() {
+    this.getBlogEntries(() => {
+      this.setState({
+        isInitialized: true
+      });
+    });
+  }
 
-    getAppliedFilterList() {
-        const appliedFilters = { ...this.state.activeFilters };
-        return Object.keys(appliedFilters).map(appliedFilterKey => ({
-            key: appliedFilterKey,
-            value: appliedFilters[appliedFilterKey]
-        }));
+  getActiveFilters() {
+    // filters are the distinct fields being returned, if there are some in the app, these should always be added to the query string
+    return this.state.filters.map(filter => filter.field);
+  }
+
+  getTopOneElement(blogEntries) {
+    let topOneElement = [];
+    topOneElement = _.slice(
+      _.orderBy(
+        blogEntries,
+        ["Pin_Blog_Entry", moment(["PublishedDate"]).format("MMM Do YY")],
+        ["desc", "asc"]
+      ),
+      0,
+      1
+    );
+    return topOneElement;
+  }
+
+  getMergeElements(topOneElement, blogEntries) {
```

Comment:
i think i know what this function is doing, but i'm not sure. Is it placing the featured post at the top of the other posts?

Reply:
yes

Reply:
yes , you are right